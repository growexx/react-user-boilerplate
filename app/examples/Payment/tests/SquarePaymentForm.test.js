import React from 'react';
import { render } from 'react-testing-library';

import SquarePaymentForm from '../SquarePaymentForm';

const inputEventType = {
  eventType: 'focusClassAdded',
  cardBrand: '',
};
const configProp = {
  applicationId: 'abcde',
  locationId: 'abcde1234',
  inputClass: 'sq-input',
  autoBuild: false,
  inputStyles: [
    {
      fontSize: '16px',
      fontFamily: 'Helvetica Neue',
      padding: '16px',
      color: '#373F4A',
      backgroundColor: 'transparent',
      lineHeight: '1.15em',
      placeholderColor: '#000',
      _webkitFontSmoothing: 'antialiased',
      _mozOsxFontSmoothing: 'grayscale',
    },
  ],
  cardNumber: {
    elementId: 'sq-card-number',
    placeholder: '• • • •  • • • •  • • • •  • • • •',
  },
  cvv: {
    elementId: 'sq-cvv',
    placeholder: 'CVV',
  },
  expirationDate: {
    elementId: 'sq-expiration-date',
    placeholder: 'MM/YY',
  },
  postalCode: {
    elementId: 'sq-postal-code',
    placeholder: 'Zip',
  },
  callbacks: {
    createPaymentRequest: () => ({
      requestShippingAddress: false,
      requestBillingInfo: true,
      currencyCode: 'USD',
      countryCode: 'US',
      total: {
        label: 'MERCHANT NAME',
        amount: 100,
        pending: false,
      },
      lineItems: [
        {
          label: 'Subtotal',
          amount: 100,
          pending: false,
        },
      ],
    }),
    cardNonceResponseReceived: (errors, nonce) => {
      if (errors) {
        this.requestPaymentSquare('');
      } else {
        this.requestPaymentSquare(nonce);
      }
    },
    inputEventReceived: (inputEvent = inputEventType) => {
      switch (inputEvent.eventType) {
        case 'focusClassAdded':
          break;
        case 'focusClassRemoved':
          break;
        case 'errorClassAdded':
          document.getElementById('error').innerHTML =
            'Please fix card information errors before continuing.';
          break;
        case 'errorClassRemoved':
          document.getElementById('error').style.display = 'none';
          break;
        case 'cardBrandChanged':
          if (inputEvent.cardBrand !== 'unknown') {
            this.setState({
              cardBrand: inputEvent.cardBrand,
            });
          } else {
            this.setState({
              cardBrand: '',
            });
          }
          break;
        case 'postalCodeChanged':
          break;
        default:
          break;
      }
    },
    paymentFormLoaded() {
      document.getElementById('name').style.display = 'inline-flex';
    },
  },
};
const props = {
  appId: '1234abc',
  location: 'aloc1234',
  amount: '100',
  requestPaymentSquare: () => {},
  PaymentForm: () => ({ build: () => {} }),
};
describe('<SquarePaymentForm />', () => {
  it('should render with appId and location', () => {
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with card nounce error', () => {
    props.PaymentForm = (config = configProp) => {
      config.callbacks.cardNonceResponseReceived({}, '');
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with card nounce', () => {
    props.PaymentForm = (config = configProp) => {
      config.callbacks.cardNonceResponseReceived(false, 'abcd1234');
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type focusClassAdded', () => {
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type focusClassRemoved', () => {
    inputEventType.eventType = 'focusClassRemoved';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type errorClassAdded', () => {
    inputEventType.eventType = 'errorClassAdded';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type errorClassRemoved', () => {
    inputEventType.eventType = 'errorClassRemoved';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type cardBrandChanged with no card', () => {
    inputEventType.eventType = 'cardBrandChanged';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type cardBrandChanged with card', () => {
    inputEventType.eventType = 'cardBrandChanged';
    inputEventType.cardBrand = 'visa';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type postalCodeChanged', () => {
    inputEventType.eventType = 'postalCodeChanged';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with event type not matched', () => {
    inputEventType.eventType = 'abcd';
    props.PaymentForm = (config = configProp) => {
      config.callbacks.inputEventReceived(inputEventType);
      return { build: () => {} };
    };
    const {
      container: { firstChild },
    } = render(<SquarePaymentForm {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
