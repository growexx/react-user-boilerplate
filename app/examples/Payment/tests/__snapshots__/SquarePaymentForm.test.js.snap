// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SquarePaymentForm /> should render with appId and location 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with card nounce 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with card nounce error 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type cardBrandChanged with card 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        >
          VISA
        </span>
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type cardBrandChanged with no card 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type errorClassAdded 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  >
    Please fix card information errors before continuing.
  </p>
</div>
`;

exports[`<SquarePaymentForm /> should render with event type errorClassRemoved 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center; display: none;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type focusClassAdded 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type focusClassRemoved 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type not matched 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;

exports[`<SquarePaymentForm /> should render with event type postalCodeChanged 1`] = `
<div
  class="container"
>
  <div
    id="form-container"
  >
    <div
      id="sq-ccbox"
    >
      <p>
        <span
          style="float: left; text-align: center;"
        >
          Enter Card Info Below 
        </span>
        <span
          style="display: block; float: right;"
        />
      </p>
      <div
        id="cc-field-wrapper"
      >
        <div
          id="sq-card-number"
        />
        <input
          id="card-nonce"
          name="nonce"
          type="hidden"
        />
        <div
          id="sq-expiration-date"
        />
        <div
          id="sq-cvv"
        />
      </div>
      <input
        id="name"
        placeholder="Name"
        style="vertical-align: top; display: none; margin: 0px; font-size: 16px; font-family: Helvetica Neue; padding: 16px; color: rgb(55, 63, 74); background-color: transparent; line-height: 1.15em;"
        type="text"
      />
      <div
        id="sq-postal-code"
      />
    </div>
    <button
      class="ant-btn button-credit-card"
      data-testid="squarePayBtn"
      type="button"
    >
      <span>
        Pay
      </span>
    </button>
  </div>
  <p
    id="error"
    style="text-align: center;"
  />
</div>
`;
